contracts/infrastructure/dapp/paraswap/ParaswapUniV2RouterFilter.sol

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2021  Argent Labs Ltd. <https://argent.xyz>
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.3;
+
+import "../BaseFilter.sol";
+import "./ParaswapUtils.sol";
+
```

Comment:
I think we should briefly describe to which contract filters will be associated to make reviewing the code easier. Something like " This contract filters calls to the <name of the contract> currently deployed at <address of the contract>".

Took me some time to remember that this contract was associated to the `UniswapProxy` contract of Paraswap and not the official `UniswapRouter`...

Reply:
BTW shouldn't the name of the contract be `ParaswapUniV3RouterFilter`?

Reply:
Agreed to add the description of the spender.

Regarding the name, I found the name given to this router by Paraswap to be too confusing. Uniswap V3 is coming out soon and this "UniswapV3Router" sounds like it's the official Uniswap V3 router when in fact it's a custom router made by Paraswap for Uniswap V2 (hence why I call it "Paraswap's UniV2 Router" i.e. `ParaswapUniV2Router`).